gap/grahom.gi

```diff
@@ -430,3 +430,122 @@ function(gr1, gr2)
   od;
   return fail;
 end);
+
+InstallMethod(IsDigraphHomomorphism, "for digraph, digraph, and perm",
+[IsDigraph, IsDigraph, IsPerm],
+function(src, ran, x)
+  local i, j;
+  if IsMultiDigraph(src) or IsMultiDigraph(ran) then
+    ErrorNoReturn("Digraphs: IsDigraphHomomorphism: usage,\n",
+                  "the first 2 arguments must not have multiple edges,");
+  elif LargestMovedPoint(x) > DigraphNrVertices(src) then
+    return false;
+  fi;
+  for i in DigraphVertices(src) do
+    for j in OutNeighbours(src)[i] do
+      if not IsDigraphEdge(ran, i ^ x, j ^ x) then
+        return false;
+      fi;
+    od;
+  od;
+  return true;
+end);
+
+InstallMethod(IsDigraphEndomorphism, "for a digraph and a perm",
+[IsDigraph, IsPerm], IsDigraphAutomorphism);
+
+InstallMethod(IsDigraphHomomorphism,
+"for digraph, digraph, and transformation",
+[IsDigraph, IsDigraph, IsTransformation],
+function(src, ran, x)
+  local i, j;
+  if IsMultiDigraph(src) or IsMultiDigraph(ran) then
+    ErrorNoReturn("Digraphs: IsDigraphHomomorphism: usage,\n",
+                  "the first 2 arguments must not have multiple edges,");
+  elif DegreeOfTransformation(x) > DigraphNrVertices(src) then
```

Comment:
Should the transformation which send 1 to 2 and fixes all other points be a digraph homomorphism between the edgeless digraph on 1 point and the edgeless digraph on 2 points? If so then there is an issue since `DegreeOfTransformation(Transformation([2,2])) = 2` but `DigraphNrVertices(Digraph([[]])) = 1` so this example will be rejected

Reply:
Could this be fixed by replacing `DegreeOfTransformation` with `LargestMovedPoint`?

Reply:
Good point @ChristopherRussell, I will fix this now.